### PR TITLE
update to 5.5.1

### PR DESCRIPTION
**Pull request by cbouss**
Changes:
- update to 5.5.1
- update dependencies

Test package on instances with GUI support.

https://github.com/spyder-ide/spyder/tree/v5.5.1

### CODE REVIEW

**Comment by SC426** [comment]
Should this be added? https://github.com/spyder-ide/spyder/blob/v5.5.1/runtests.py

File: recipe/meta.yaml [hunk 1]
```diff
@@ -20,7 +19,8 @@ requirements:
   host:
     - python
     - pip
-    - setuptools
+    - setuptools >=61.2.0
```
Comment (danpetry) [inline recipe/meta.yaml:22]: these setuptools requirements seem to have come from the python-lsp-server repo, which is copied in

File: recipe/meta.yaml [hunk 2]
```diff
@@ -47,24 +48,24 @@ requirements:
    - psutil >=5.3
    - pygments >=2.0
    - pylint >=2.5.0,<3.0
```
Comment (danpetry) [inline recipe/meta.yaml:50]:
```suggestion
   - pylint >=2.5.0,<3.1
```

**Review by danpetry — COMMENTED** [review]
I guess the comment means the GUI has been tested?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
 
 source:
   url: https://pypi.io/packages/source/s/spyder/spyder-{{ version }}.tar.gz
-  sha256: 14c86757a5122cb0785817d0ab6c00d8fe894ebd214c97e8ef5c44f962601371
+  sha256: fb3f098f4780fe6607d6bf19414c98505324ee1d660718e84c3e58664d1c1f49
 
 build:
-  number: 1
+  number: 0
   skip: true  # [ppc64le or s390x or py<38]
```
[the remark this answers is not in the document]
Comment (cbouss) [reply recipe/meta.yaml]: addressed

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: addressed

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [reply recipe/meta.yaml]: addressed

**Comment by skupr-anaconda** [comment]
The test environment for py39 on win64 failed. Is it something bad on Prefect?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -46,25 +46,25 @@ requirements:
    - ptyprocess >=0.5  # [win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:46]: I wonder if it's still relevant. See https://github.com/spyder-ide/spyder/pull/16766#issuecomment-963601853 and https://github.com/conda-forge/spyder-feedstock/pull/119

Comment (cbouss) [reply recipe/meta.yaml:46]: maybe not, but I'll leave the research effort for the next update.

**Comment by cbouss** [comment]
> The test environment for py39 on win64 failed. Is it something bad on Prefect?

I think so. I'll restart the flow.

**Comment by cbouss** [comment]
I tested the application again on GUI. It opens fine. Putting back for review.

**Review by skupr-anaconda — APPROVED** [review]

**Review by danpetry — APPROVED** [review]

**Pull request merged**